### PR TITLE
Refactor metric PipelineAggregation integration test

### PR DESCRIPTION
**Pull request by iverase**
Follow up https://github.com/elastic/elasticsearch/pull/77481, there are several pipeline aggregation integration test that seems to have been cloned and can easily be refactor in a test case.

This PR extract `BucketMetricsPipeLineAggregationTestCase` and simplifies the testing of several pipeline aggregations.

### CODE REVIEW

**Review by nik9000 — COMMENTED** [review]
This is great!

File: server/src/internalClusterTest/java/org/elasticsearch/search/aggregations/pipeline/BucketMetricsPipeLineAggregationTestCase.java [hunk 1]
```diff
@@ -0,0 +1,494 @@
...
+     * The bug was caused by an UnmappedTerms being the chosen as the first reduction target.  UnmappedTerms
+     * delegated reduction to the first non-unmapped agg, which would reduce and run pipeline aggs.  But then
+     * execution returns to the UnmappedTerms and _it_ runs pipelines as well, doubling up on the values.
+     */
+    public void testFieldIsntWrittenOutTwice() throws Exception {
+        // you need to add an additional index with no fields in order to trigger this (or potentially a shard)
+        // so that there is an UnmappedTerms in the list to reduce.
+        createIndex("foo_1");
+
+        XContentBuilder builder = jsonBuilder().startObject()
```
[the remark this answers is not in the document]
Comment (iverase) [reply server/src/internalClusterTest/java/org/elasticsearch/search/aggregations/pipeline/BucketMetricsPipeLineAggregationTestCase.java:444]: umm, yes this is how spotless has formatted this code. What do you mean to fix the formatting? Once we are using spotless, I think it has the last word on 'correct' formatting :)

[the remark this answers is not in the document]
Comment (not-napoleon) [reply server/src/internalClusterTest/java/org/elasticsearch/search/aggregations/pipeline/BucketMetricsPipeLineAggregationTestCase.java:444]: You can disable spotless for blocks or whole files via directives in the comments: `// @formatter:off` and `// @formatter:on`.  We shouldn't overuse it, but formatting XContentBuilder is a good place for it.

[the remark this answers is not in the document]
Comment (iverase) [reply server/src/internalClusterTest/java/org/elasticsearch/search/aggregations/pipeline/BucketMetricsPipeLineAggregationTestCase.java:444]: Thanks @not-napoleon, I formatted this bit of code.

[the remark this answers is not in the document]
Comment (nik9000) [reply server/src/internalClusterTest/java/org/elasticsearch/search/aggregations/pipeline/BucketMetricsPipeLineAggregationTestCase.java:444]: I was thinking of splitting it into multiple lines or something. I dunno about spotless being the last word on formatting - if the code is hard to read we should do *something*.

**Comment by iverase** [comment]
ping @not-napoleon @nik9000 what needs to be done here?

**Review by nik9000 — APPROVED** [review]
I think it's good. Sorry, I hadn't noticed the changes.

**Review by not-napoleon — APPROVED** [review]
I'm also good with this if Nik is.

**Pull request merged**